src/zenml/integrations/hyperai/orchestrators/hyperai_orchestrator.py

```diff
@@ -0,0 +1,445 @@
+                    "`ssh_key_type` of its connected service connector is set to the "
+                    "correct value."
+                )
+            elif not isinstance(paramiko_client, paramiko.SSHClient):
+                raise RuntimeError(
+                    f"Expected to receive a `paramiko.SSHClient` object from the "
+                    f"linked connector, but got type `{type(paramiko_client)}`."
+                )
+        else:
+            raise RuntimeError(
+                "You must link a HyperAI service connector to the orchestrator."
+            )
+
+        # Get container registry autologin setting
+        container_registry_autologin = self.config.container_registry_autologin
+        if container_registry_autologin:
+            logger.info(
+                "Attempting to automatically log in to container registry used by stack."
+            )
+
+            # Select stack container registry
+            container_registry = None
+            for component in stack.components.values():
+                if isinstance(component, BaseContainerRegistry):
+                    container_registry = component
+                    break
+
+            # Raise error if no container registry is found
+            if not container_registry:
+                raise RuntimeError(
+                    "Unable to find container registry in stack."
+                )
+
+            # Get container registry credentials from its config
+            container_registry_url = container_registry.config.uri
+            (
+                container_registry_username,
+                container_registry_password,
+            ) = container_registry.credentials
+
+            # Log in to container registry using --password-stdin
+            stdin, stdout, stderr = paramiko_client.exec_command(
+                f"docker login -u {container_registry_username} --password-stdin {container_registry_url} <<< {container_registry_password}"
+            )
+
+            # Log stdout
+            for line in stdout.readlines():
+                logger.info(line)
+
+        # Set up pipeline-runs directory if it doesn't exist
+        nonscheduled_directory_name = "/home/zenml/pipeline-runs"
+        directory_name = (
+            nonscheduled_directory_name
+            if not deployment.schedule
+            else "/home/zenml/scheduled-pipeline-runs"
+        )
+        stdin, stdout, stderr = paramiko_client.exec_command(
+            f"mkdir -p {directory_name}"
+        )
+
+        # Get pipeline run id and create directory for it
+        orchestrator_run_id = self.get_orchestrator_run_id()
+        directory_name = f"{directory_name}/{orchestrator_run_id}"
+        stdin, stdout, stderr = paramiko_client.exec_command(
+            f"mkdir -p {directory_name}"
+        )
+
+        # Remove all folders from nonscheduled pipelines if they are 7 days old or older
+        stdin, stdout, stderr = paramiko_client.exec_command(
```

Comment:
Is this maybe something that we should make configurable?

Reply:
We can, yes. But what do you think about setting it to `True` by default? It is all too common to forget cleanup, and although these are small files, they can pile up quickly.

Reply:
![image](https://github.com/zenml-io/zenml/assets/6450612/1da933ab-b7bd-486c-bc0c-83fa08d94e6a)

Added and works!